com.avaloq.tools.ddk.xtext/src/com/avaloq/tools/ddk/xtext/extension/ILanguageExtensionsService.java

```diff
@@ -0,0 +1,38 @@
+/*******************************************************************************
+ * Copyright (c) 2016 Avaloq Evolution AG and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Avaloq Evolution AG - initial API and implementation
+ *******************************************************************************/
+
+package com.avaloq.tools.ddk.xtext.extension;
+
+import java.util.Collection;
+
+import com.google.inject.Injector;
+
+
+/**
+ * Global service that manages all extensions to all languages.
+ * <p>
+ * This service is implemented as a global singleton, one instance for all DSLs.

```

Comment:
Well, strictly speaking this is not quite the case. The "scope" of singleton in guice is the Injector. Since we have injectors for all DSLs, there will be an instance per language